### PR TITLE
[Drop-In UI] Added missing Binder unit tests

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Closes NAVAND-703

### Description

Added unit tests for following Binder classes:
- ViewBinder
- EmptyBinder
- SpeedLimitViewBinder
- RoadNameViewBinder
- ManeuverViewBinder
- InfoPanelHeaderDestinationPreviewBinder
- InfoPanelHeaderArrivalBinder
- InfoPanelHeaderRoutesPreviewBinder
- InfoPanelHeaderActiveGuidanceBinder
- InfoPanelEndNavigationButtonBinder
- MapBinder

### CODE REVIEW

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+            if (shouldBind) {
+                assertNotNull("Should attach RouteArrowComponent for $navigationState", c)
+            } else {
+                assertNull("Should not attach RouteArrowComponent for $navigationState", c)
+            }
+        }
+    }
+
+    @Test
+    fun `bind should reload RouteArrowComponent on mapStyle change`() {
```
Comment (dzinad) [inline libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt:336]: It should also reload on routeArrowOptions change.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            assertNull(reloadComponent.childComponent)
+
+            loadedMapStyleFlow.value = mockk()
+
+            assertNotNull(reloadComponent.childComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on roadNameBackground change`() =
```
Comment (dzinad) [inline libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt:111]: We can also check that background is applied to the view.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            navContext.applyStyleCustomization {
+                roadNameBackground = R.drawable.mapbox_bg_button_primary
+            }
+
+            assertNotEquals(firstComponent, reloadComponent.childComponent)
+        }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should return ReloadingComponent that reloads child component on roadNameTextAppearance change`() =
```
Comment (dzinad) [inline libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt:127]: We can also check that text appearance is applied.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/infopanel/InfoPanelHeaderActiveGuidanceBinderTest.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        sut = InfoPanelHeaderActiveGuidanceBinder(navContext)
+    }
+
+    @Test
+    fun `bind should inflate active guidance header layout`() {
+        val rootLayout = FrameLayout(ctx)
+
+        sut.bind(rootLayout)
+
+        assertNotNull(rootLayout.findViewById(R.id.tripProgressLayout))
```
[the remark this answers is not in the document]
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/infopanel/InfoPanelHeaderActiveGuidanceBinderTest.kt:65]: I think that would be overkill. These two assertions are good enough to verify that the correct layout has been inflated.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(
+    ExperimentalPreviewMapboxNavigationAPI::class,
+    ExperimentalCoroutinesApi::class,
+    FlowPreview::class
+)
+@RunWith(RobolectricTestRunner::class)
+class MapBinderTest {
```
[the remark this answers is not in the document]
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt:69]: I know. But don't worry about it. I'll help resolve any conflicts caused by these changes.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/infopanel/InfoPanelEndNavigationButtonBinderTest.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    private lateinit var ctx: Context
+    private lateinit var navContext: NavigationViewContext
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var sut: InfoPanelEndNavigationButtonBinder
+
+    @Before
+    fun setUp() {
+        ctx = ApplicationProvider.getApplicationContext()
+        navContext = spyk(
```
[the remark this answers is not in the document]
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/infopanel/InfoPanelEndNavigationButtonBinderTest.kt:44]: Here... Copy/Paste. Writing so many unit test classes involves a lot of copying and pasting of the same test harness setup.
Spy is great. It lets you use real objects and allows you verification (or alteration) of method calls.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt (hunk 3 above)
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt:127]: We could but I think we should first move that responsibility to RoadNameComponent.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt (hunk 2 above)
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/roadname/RoadNameViewBinderTest.kt:111]: Let's first move that responsibility to RoadNameComponent and then add a test for it.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt (hunk 1 above)
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt:337]: I cover that in a separate test https://github.com/mapbox/mapbox-navigation-android/pull/6446/files#diff-dcc047be08034dcd81128312c2da7ea914b85cb3f77587c85afaaa3e91e35d91R294
Line 294.

File: libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/infopanel/InfoPanelEndNavigationButtonBinderTest.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (tomaszrybakiewicz) [reply libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/infopanel/InfoPanelEndNavigationButtonBinderTest.kt]: Updated. I only need Spies for tests that need a mock mapStyleLoader.

**Review by dzinad — APPROVED** [review]

**Pull request merged**